backend/coins/btc/synchronizer/synchronizer.go

```diff
@@ -70,13 +70,13 @@ func (synchronizer *Synchronizer) decRequestsCounter() {
 
 // WaitSynchronized blocks until all pending synchronization tasks are finished.
 func (synchronizer *Synchronizer) WaitSynchronized() {
-	synchronizer.log.WithFields(logrus.Fields{"requestCounter": synchronizer.requestsCounter}).
-		Debug("wait synchronized")
-	if func() int32 {
-		defer synchronizer.waitLock.RLock()()
-		return synchronizer.requestsCounter
-	}() == 0 {
+
+	unlock := synchronizer.waitLock.RLock()
+	n := synchronizer.requestsCounter
+	unlock()
+	if n == 0 {
 		return
 	}
+	synchronizer.log.WithFields(logrus.Fields{"requestCounter": n}).Debug("wait synchronized")
```

Comment:
this increases log spam by a lot?

Reply:
it looks that way. from a support ticket:

```
$ wc -l log.txt
3280 log.txt
$ grep "wait synchronized" log.txt | wc -l
2229
```

almost 68% is that line.

Reply:
ah i see, it's moved to after `==0`. obvoiusly `requestsCounter` here would always be 0, so that is usless info.

i'd just delete the whole log here. it has never been useful in the past.

Reply:
yeah, I moved things a bit here because `synchronizer.requestsCounter` should be accessed under a lock, from what I see, and the logging was actually done while still locked.

Reply:
> while still locked.

I mean while still **unlocked**.

Reply:
> i'd just delete the whole log here

done